### PR TITLE
[Release] Grep App

### PR DESCRIPTION
**Pull request by syumet**
[Ticket](https://www.notion.so/0f23d2c25e624e7faf36675d3e64dd1f?v=e3f9938f22934f6b9b4ccbd4fd5f95f7&p=229ba0a35d5944bba8b0bd04327bb260)

- [Initialize core_java project](https://www.notion.so/38ed2bcfc93b421b92de84e5d6f728c3?v=08ebea1ef0a4457197e2910f2d76bcd2&p=f19b0717fc1c45bd93473a26e976d26f)

- [Java RegexR](https://www.notion.so/38ed2bcfc93b421b92de84e5d6f728c3?v=08ebea1ef0a4457197e2910f2d76bcd2&p=2b222ba5b47d47ddbcc28fba37b38c88)

- [Design & Implementation](https://www.notion.so/38ed2bcfc93b421b92de84e5d6f728c3?v=08ebea1ef0a4457197e2910f2d76bcd2&p=5519caba1e3645e88dc112572ea4b7b7)

- [Learning Lambda and Stream APIs](https://www.notion.so/38ed2bcfc93b421b92de84e5d6f728c3?v=08ebea1ef0a4457197e2910f2d76bcd2&p=573e3b4a33014bb6be6890784ce61aba)

- [Leambda and Stream API Implmentation](https://www.notion.so/38ed2bcfc93b421b92de84e5d6f728c3?v=08ebea1ef0a4457197e2910f2d76bcd2&p=0c98340332b14e05b54ef33803bbab50)

- [Readme](https://www.notion.so/38ed2bcfc93b421b92de84e5d6f728c3?v=08ebea1ef0a4457197e2910f2d76bcd2&p=b04739190ef54bca8655699b1940fed6)

### CODE REVIEW

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+  public JavaGrepImp() {
+  }
+
+  public static void main(String[] args) {
+    if (args.length != 3) {
+      throw new IllegalArgumentException("Usage: JavaGrep [regex] [rootPath] [outFilePath]");
+    }
+    JavaGrepImp grep = new JavaGrepImp(args[0], args[1], args[2]);
+    BasicConfigurator.configure(); // configure log4j
```
Comment (jarviscanada) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java]: wrong comment format

Comment (jarviscanada) [reply core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java]: you can remove line 44.

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    try {
+      Files.walkFileTree(Paths.get(rootDir), new SimpleFileVisitor<Path>() {
+        @Override
+        public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+          ls.add(file.toFile());
+          return super.visitFile(file, attrs);
+        }
+      });
+    } catch (IOException e) {
+      this.logger.error(e.getMessage(), e);
```
Comment (jarviscanada) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java]: by convention, you don't need `this.`. please remove

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImp.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    writeToFile(
+        listFiles(getRootPath())
+            .stream()
+            .flatMap(
+                f -> readLines(f)
+                    .stream()
+                    .filter(this::containsPattern)
+            )
+            .collect(Collectors.toList())
+    ); // looks cool but is it really better than super's ?
```
Comment (jarviscanada) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImp.java]: wrong comment format (comments are always above code.

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImp.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    }
+    return ls;
+  }
+
+  @Override
+  public List<String> readLines(File inputFile) {
+    try (Stream<String> stream = Files.lines(inputFile.toPath())) {
+      return stream.collect(Collectors.toList());
+    } catch (IOException e) {
+      e.printStackTrace();
```
Comment (jarviscanada) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImp.java]: don't printStackTrace

File: practice/src/main/java/ca/jrvs/practice/dataStructure/HelloWorld.java [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package ca.jrvs.practice.dataStructure;
+
+class HelloWorld {
+
```
Comment (jarviscanada) [inline practice/src/main/java/ca/jrvs/practice/dataStructure/HelloWorld.java]: remove this

**Review by jarviscanada — APPROVED** [review]

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java (hunk 1 above)
Comment (syumet) [reply core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java]: line 44 is linked to [this issue](https://jarvis-training.slack.com/archives/CUNGTFZGQ/p1584977723026000) I had

Comment (syumet) [reply core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java]: interestingly I find that I don't need it right now...but I do have this issue at the first time

**Comment by syumet** [comment]
comments resolved

**Pull request merged**